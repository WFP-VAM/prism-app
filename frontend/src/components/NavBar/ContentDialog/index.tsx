import React, { memo, useEffect, useRef, useState } from 'react';
import parse from 'html-react-parser';
import { marked } from 'marked';
import { Dialog, DialogContent, Typography } from '@material-ui/core';

type ContentDialogProps = {
  content: string;
  handleClose: () => void;
  elementId?: string;
};

const ContentDialog = memo(
  ({ content, handleClose, elementId }: ContentDialogProps) => {
    const [open, setOpen] = useState(false);

    const contentRef = useRef<HTMLDivElement>(null);

    useEffect(() => {
      if (!content) {
        setOpen(false);
        return;
      }
      setOpen(true);
    }, [content]);

    useEffect(() => {
      if (!open || !elementId) {
        return;
      }
      (document.querySelector(
        `#${elementId}`,
      ) as HTMLHeadingElement).scrollIntoView({
        behavior: 'smooth',
        block: 'start',
      });
    }, [elementId, open]);

    return (
      <Dialog
        maxWidth="md"
        open={open}
        keepMounted
        onClose={handleClose}
        aria-labelledby="dialog-preview"
      >
        <DialogContent>
          <Typography ref={contentRef} component="div" color="textSecondary">
            {parse(marked(content))}
          </Typography>
        </DialogContent>
      </Dialog>
    );
  },
);

export default ContentDialog;
