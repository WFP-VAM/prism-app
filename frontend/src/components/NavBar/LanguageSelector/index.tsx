import React from 'react';
import {
  Button,
  ButtonGroup,
  createStyles,
  Theme,
  Typography,
  withStyles,
  WithStyles,
} from '@material-ui/core';
import { languages, useSafeTranslation } from 'i18n';

function LanguageSelector({ classes }: LanguageSelectorProps) {
  const { i18n } = useSafeTranslation();

  const handleChangeLanguage = (lng: string): void => {
    i18n.changeLanguage(lng);
  };
  // If there is only one language, hide the selector
  if (languages.length <= 1) {
    return null;
  }

  return (
    <ButtonGroup variant="text" className={classes.block}>
      {languages.map(lng => (
        <Button
          key={lng}
          type="submit"
          onClick={() => handleChangeLanguage(lng)}
        >
          <Typography
            variant="body2"
            style={{
              fontWeight: i18n.resolvedLanguage === lng ? 'bold' : 'normal',
            }}
          >
            {lng}
          </Typography>
        </Button>
      ))}
    </ButtonGroup>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    block: {
      paddingLeft: '10px',
      paddingTop: '4px',
    },
    title: {
      color: theme.palette.text.secondary,
    },
  });

export interface LanguageSelectorProps extends WithStyles<typeof styles> {}

export default withStyles(styles)(LanguageSelector);
