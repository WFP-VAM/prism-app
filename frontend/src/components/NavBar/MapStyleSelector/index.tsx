import React from 'react';
import {
  Button,
  createStyles,
  makeStyles,
  Menu,
  MenuItem,
  Typography,
} from '@material-ui/core';
import { useSafeTranslation } from 'i18n';
import { appConfig } from 'config';
import { get } from 'lodash';
import { ArrowDropDown, Public } from '@material-ui/icons';
import { useDispatch, useSelector } from 'react-redux';
import { mapStyleSelector } from 'context/mapStateSlice/selectors';
import { MapStyle, setMapStyle } from 'context/mapStateSlice';

const mapStyles: MapStyle[] = get(appConfig.map, 'styles', []);

function MapStyleSelector() {
  const classes = useStyles();
  const dispatch = useDispatch();
  const { t } = useSafeTranslation();
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const mapStyle = useSelector(mapStyleSelector);

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const handleChange = (style: MapStyle): void => {
    dispatch(setMapStyle(style));
    handleClose();
  };

  if (mapStyles.length <= 1) {
    return null;
  }

  return (
    <>
      <Button
        aria-label="map-style-select-dropdown-button"
        style={{ paddingLeft: 0 }}
        onClick={handleClick}
        endIcon={<ArrowDropDown fontSize="small" />}
      >
        <Public style={{ paddingRight: '0.5rem' }} />
        <Typography color="secondary" style={{ textTransform: 'none' }}>
          {t(mapStyle?.label || '')}
        </Typography>
      </Button>
      <Menu
        open={Boolean(anchorEl)}
        onClose={handleClose}
        className={classes.block}
        anchorEl={anchorEl}
      >
        {mapStyles?.map(x => (
          <MenuItem key={x.id} onClick={() => handleChange(x)}>
            <Typography>{t(x.label)}</Typography>
          </MenuItem>
        ))}
      </Menu>
    </>
  );
}

const useStyles = makeStyles(() =>
  createStyles({
    block: {
      paddingLeft: '10px',
      paddingTop: '4px',
    },
  }),
);

export default MapStyleSelector;
