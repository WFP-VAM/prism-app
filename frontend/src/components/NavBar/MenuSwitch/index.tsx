import React from 'react';
import { useDispatch } from 'react-redux';
import {
  Button,
  createStyles,
  Grid,
  Typography,
  withStyles,
  WithStyles,
} from '@material-ui/core';
import { LayersCategoryType, TableType } from '../../../config/types';
import { loadTable } from '../../../context/tableStateSlice';
import { useSafeTranslation } from '../../../i18n';
import SwitchItem from './SwitchItem';

function MenuSwitch({ classes, title, layers, tables }: MenuSwitchProps) {
  const { t } = useSafeTranslation();
  const dispatch = useDispatch();

  const showTableClicked = (table: TableType) => {
    dispatch(loadTable(table.id));
  };

  return (
    <Grid item key={title} className={classes.categoryContainer}>
      <Typography variant="body2" className={classes.categoryTitle}>
        {t(title)}
      </Typography>
      <hr />

      {layers.map(layer => {
        if (
          layer.group &&
          layer.group.layers.find(l => l.id === layer.id && !l.main)
        ) {
          return null;
        }
        return <SwitchItem key={layer.id} layer={layer} />;
      })}

      {tables.map(table => (
        <Button
          id={table.title}
          key={table.title}
          onClick={() => showTableClicked(table)}
        >
          <Typography variant="body1">{table.title}</Typography>
        </Button>
      ))}
    </Grid>
  );
}

const styles = () =>
  createStyles({
    categoryContainer: {
      marginBottom: 16,
      '&:last-child': {
        marginBottom: 0,
      },
    },
    categoryTitle: {
      fontWeight: 'bold',
      textAlign: 'left',
    },
  });

export interface MenuSwitchProps
  extends LayersCategoryType,
    WithStyles<typeof styles> {}

export default withStyles(styles)(MenuSwitch);
