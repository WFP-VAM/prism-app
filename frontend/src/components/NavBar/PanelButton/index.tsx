import { Button, IconButton, Badge, Typography } from '@material-ui/core';
import React from 'react';
import { ExpandMore } from '@material-ui/icons';
import { black, cyanBlue } from 'muiTheme';
import useLayers from 'utils/layers-utils';
import { useSelector } from 'react-redux';
import { analysisResultSelector } from 'context/analysisResultStateSlice';
import { Panel } from 'config/types';

function PanelButton({
  panel,
  selected,
  handleClick,
  isMobile,
  buttonText,
}: {
  panel: any;
  selected: boolean;
  handleClick: (event: React.MouseEvent<HTMLElement>) => void;
  isMobile: boolean;
  buttonText: string;
}) {
  const { numberOfActiveLayers } = useLayers();
  const analysisData = useSelector(analysisResultSelector);
  const badgeContent = numberOfActiveLayers + Number(Boolean(analysisData));
  const Wrap =
    badgeContent >= 1 && panel.panel === Panel.Layers
      ? // eslint-disable-next-line react/no-unused-prop-types
        ({ children }: { children: React.ReactNode }) => (
          <Badge
            anchorOrigin={{
              horizontal: 'left',
              vertical: 'top',
            }}
            overlap="rectangular"
            badgeContent={badgeContent}
            color="secondary"
          >
            {children}
          </Badge>
        )
      : ({ children }: { children: React.ReactNode }) => children;

  const commonStyles = {
    backgroundColor: selected ? cyanBlue : undefined,
    color: selected ? black : 'white',
  };

  const renderButtonContent = (
    <Typography
      style={{
        color: selected ? black : '#FFFF',
        textTransform: 'none',
      }}
    >
      {buttonText}
    </Typography>
  );

  return isMobile ? (
    <Wrap>
      <IconButton style={commonStyles} onClick={handleClick}>
        {panel.icon}
      </IconButton>
    </Wrap>
  ) : (
    <Button
      style={commonStyles}
      startIcon={<Wrap>{panel.icon}</Wrap>}
      endIcon={panel.children ? <ExpandMore fontSize="small" /> : null}
      onClick={handleClick}
    >
      {renderButtonContent}
    </Button>
  );
}

export default PanelButton;
