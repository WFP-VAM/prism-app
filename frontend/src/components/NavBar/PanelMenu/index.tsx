import { Menu, MenuItem } from '@material-ui/core';
import { PanelItem } from 'config/types';

function PanelMenu({
  panel,
  menuAnchor,
  handleMenuClose,
  handleChildClick,
  selected,
}: {
  panel: PanelItem;
  menuAnchor: HTMLElement | null;
  handleMenuClose: () => void;
  handleChildClick: (childPanel: PanelItem) => void;
  selected: string;
}) {
  const validSelected = panel.children?.find(
    (child: PanelItem) => child.panel === selected,
  );

  return (
    <Menu
      anchorEl={menuAnchor}
      open={Boolean(menuAnchor)}
      onClose={handleMenuClose}
    >
      {panel.children?.map((child: PanelItem) => (
        <MenuItem
          key={child.panel}
          onClick={() => {
            handleChildClick(child);
            handleMenuClose();
          }}
          selected={validSelected?.panel === child.panel}
        >
          {child.label}
        </MenuItem>
      ))}
    </Menu>
  );
}

export default PanelMenu;
