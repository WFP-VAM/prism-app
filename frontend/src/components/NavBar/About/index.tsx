import React, { useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Typography, Grid } from '@material-ui/core';
import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
import { appConfig } from '../../../config';
import ContentDialog, { loadLayerContent } from '../ContentDialog';
import { useSafeTranslation } from '../../../i18n';

const About = () => {
  const [content, setContent] = useState<string | undefined>(undefined);
  const { t } = useSafeTranslation();
  const { aboutPath } = appConfig;

  const href = aboutPath ? '' : 'https://innovation.wfp.org/project/prism';
  const handler = aboutPath
    ? (event: any) => {
        event.preventDefault();
        loadLayerContent(aboutPath, setContent);
      }
    : undefined;

  return (
    <Grid item key="About">
      <Typography
        variant="body2"
        component="a"
        target="_blank"
        href={href}
        onClick={handler}
      >
        <FontAwesomeIcon icon={faInfoCircle} /> {t('About')}
      </Typography>
      <ContentDialog content={content} setContent={setContent} />
    </Grid>
  );
};

export default About;
