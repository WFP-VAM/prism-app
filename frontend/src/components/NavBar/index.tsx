/* eslint-disable react/prop-types */
import { faGithub } from '@fortawesome/free-brands-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  AppBar,
  Box,
  createStyles,
  Theme,
  Toolbar,
  Typography,
  IconButton,
  makeStyles,
  useTheme,
  useMediaQuery,
} from '@material-ui/core';
import React, { useState } from 'react';
import { useSafeTranslation } from 'i18n';
import { appConfig } from 'config';
import {
  BarChartOutlined,
  ImageAspectRatioOutlined,
  LayersOutlined,
  TableChartOutlined,
  TimerOutlined,
  Notifications,
} from '@material-ui/icons';
import { useDispatch, useSelector } from 'react-redux';
import {
  leftPanelTabValueSelector,
  setTabValue,
} from 'context/leftPanelStateSlice';
import GoToBoundaryDropdown from 'components/Common/BoundaryDropdown/goto';
import Legends from 'components/MapView/Legends';
import { areChartLayersAvailable } from 'config/utils';
import {
  areTablesAvailable,
  isAnticipatoryActionDroughtAvailable,
  isAnticipatoryActionStormAvailable,
} from 'components/MapView/LeftPanel/utils';
import { Panel, PanelItem } from 'config/types';
import About from './About';
import LanguageSelector from './LanguageSelector';
import PrintImage from './PrintImage';
import PanelMenu from './PanelMenu';
import PanelButton from './PanelButton';

const { alertFormActive, header } = appConfig;

const panels: PanelItem[] = [
  { panel: Panel.Layers, label: 'Layers', icon: <LayersOutlined /> },
  ...(areChartLayersAvailable
    ? [{ panel: Panel.Charts, label: 'Charts', icon: <BarChartOutlined /> }]
    : []),
  {
    panel: Panel.Analysis,
    label: 'Analysis',
    icon: <ImageAspectRatioOutlined />,
  },
  ...(areTablesAvailable
    ? [{ panel: Panel.Tables, label: 'Tables', icon: <TableChartOutlined /> }]
    : []),
  ...(isAnticipatoryActionDroughtAvailable || isAnticipatoryActionStormAvailable
    ? [
        {
          label: 'A. Actions',
          icon: <TimerOutlined />,
          children: [
            ...(isAnticipatoryActionDroughtAvailable
              ? [
                  {
                    panel: Panel.AnticipatoryActionDrought,
                    label: 'A. Action Drought',
                    icon: <TimerOutlined />,
                  },
                ]
              : []),
            ...(isAnticipatoryActionStormAvailable
              ? [
                  {
                    panel: Panel.AnticipatoryActionStorm,
                    label: 'A. Action Storm',
                    icon: <TimerOutlined />,
                  },
                ]
              : []),
          ],
        },
      ]
    : []),
  ...(alertFormActive
    ? [{ panel: Panel.Alerts, label: '', icon: <Notifications /> }]
    : []),
];

function NavBar() {
  const { t } = useSafeTranslation();
  const dispatch = useDispatch();
  const classes = useStyles();
  const tabValue = useSelector(leftPanelTabValueSelector);
  const theme = useTheme();
  const smDown = useMediaQuery(theme.breakpoints.down('sm'));
  const mdUp = useMediaQuery(theme.breakpoints.up('md'));
  const [menuAnchor, setMenuAnchor] = useState<{
    [key: string]: HTMLElement | null;
  }>({});
  const [selectedChild, setSelectedChild] = useState<Record<string, PanelItem>>(
    {},
  );

  const rightSideLinks = [
    {
      title: 'GitHub',
      icon: faGithub,
      href: 'https://github.com/wfp-VAM/prism-app',
    },
  ];

  const buttons = rightSideLinks.map(({ title, icon, href }) => (
    <IconButton
      key={title}
      component="a"
      target="_blank"
      href={href}
      style={{ color: 'white' }}
    >
      <FontAwesomeIcon fontSize={mdUp ? '1.25rem' : '1.5rem'} icon={icon} />
    </IconButton>
  ));

  const handleMenuOpen = (
    key: string,
    event: React.MouseEvent<HTMLElement>,
  ) => {
    setMenuAnchor(prev => ({ ...prev, [key]: event.currentTarget }));
  };

  const handleMenuClose = (key: string) => {
    setMenuAnchor(prev => ({ ...prev, [key]: null }));
  };

  const handlePanelClick = (panel: Panel) => {
    dispatch(setTabValue(panel));
  };

  const handleChildSelection = (panel: any, child: any) => {
    setSelectedChild({
      [panel.label]: child,
    });
    handleMenuClose(panel.label);
    handlePanelClick(child.panel);
  };

  const { title, subtitle, logo } = header || {
    title: 'PRISM',
  };

  return (
    <AppBar position="static" className={classes.appBar}>
      <Toolbar variant="dense">
        <div className={classes.navbarContainer}>
          <div className={classes.leftSideContainer}>
            <div className={classes.titleContainer}>
              {logo && <img className={classes.logo} src={logo} alt="logo" />}
              <Box
                style={{
                  display: 'flex',
                  flexDirection: 'column',
                }}
              >
                {title && (
                  <Typography
                    color="secondary"
                    variant="h6"
                    className={classes.title}
                  >
                    {t(title)}
                  </Typography>
                )}
                {subtitle && !smDown && (
                  <Typography
                    color="secondary"
                    variant="subtitle2"
                    className={classes.subtitle}
                  >
                    {t(subtitle)}
                  </Typography>
                )}
              </Box>
            </div>
            <div className={classes.panelsContainer}>
              {panels.map(panel => {
                const selected =
                  tabValue === panel.panel ||
                  (panel.children &&
                    panel.children.some(child => tabValue === child.panel));

                const buttonText = selectedChild[panel.label]
                  ? selectedChild[panel.label].label
                  : t(panel.label);

                return (
                  <React.Fragment key={panel.panel}>
                    <PanelButton
                      panel={panel}
                      selected={selected || false}
                      handleClick={e => {
                        if (panel.children) {
                          handleMenuOpen(panel.label, e);
                        } else if (panel.panel) {
                          handlePanelClick(panel.panel);
                        }
                      }}
                      isMobile={!mdUp}
                      buttonText={buttonText}
                    />
                    {panel.children && (
                      <PanelMenu
                        panel={panel}
                        menuAnchor={menuAnchor[panel.label]}
                        handleMenuClose={() => handleMenuClose(panel.label)}
                        handleChildClick={(child: any) =>
                          handleChildSelection(panel, child)
                        }
                        selected={tabValue}
                      />
                    )}
                  </React.Fragment>
                );
              })}
              <GoToBoundaryDropdown />
            </div>
          </div>
          <div className={classes.rightSideContainer}>
            <Legends />
            <PrintImage />
            {buttons}
            <About />
            <LanguageSelector />
          </div>
        </div>
      </Toolbar>
    </AppBar>
  );
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    logo: {
      height: 32,
      marginRight: 15,
    },
    appBar: {
      backgroundImage: `linear-gradient(180deg, ${theme.palette.primary.main} 0%, ${theme.palette.primary.dark} 100%)`,
      height: '6vh',
      display: 'flex',
      justifyContent: 'center',
    },
    panelButton: {
      height: '2.5em',
    },
    title: {
      letterSpacing: '.3rem',
      fontSize: '1.25rem',
      lineHeight: '1.5rem',
      textTransform: 'uppercase',
      padding: 0,
    },
    subtitle: {
      fontSize: '.8rem',
      fontWeight: 300,
      letterSpacing: '.1rem',
      lineHeight: '.8rem',
      padding: 0,
    },
    menuContainer: {
      textAlign: 'center',
    },
    mobileDrawerContent: {
      backgroundColor: theme.palette.primary.main,
      paddingTop: 16,
      width: '80vw',
      height: '100vh',
      overflowX: 'hidden',
      display: 'flex',
      flexDirection: 'column',
      paddingLeft: '1em',
    },
    menuBars: {
      height: '100%',
      fontSize: 20,
      color: theme.palette.text.primary,
    },
    mobileMenuContainer: {
      textAlign: 'right',
    },
    navbarContainer: {
      display: 'flex',
      justifyContent: 'space-between',
      flexDirection: 'row',
      width: '100%',
    },
    leftSideContainer: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'start',
      gap: '5rem',
    },
    titleContainer: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'start',
      gap: '1rem',
      alignItems: 'center',
    },
    panelsContainer: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'start',
      gap: '1rem',
      alignItems: 'center',
    },
    rightSideContainer: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'start',
      gap: '0.5rem',
      alignItems: 'center',
    },
  }),
);

export default NavBar;
