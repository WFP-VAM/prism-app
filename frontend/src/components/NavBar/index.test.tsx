import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import NavBar from '.';
import { store } from '../../context/store';

jest.mock('./PrintImage', () => 'mock-PrintImage');

test('renders as expected', () => {
  const { container } = render(
    <Provider store={store}>
      <NavBar />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
