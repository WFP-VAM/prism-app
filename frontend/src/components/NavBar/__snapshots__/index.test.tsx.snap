// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-appbar
    classname="makeStyles-appBar-2"
    position="static"
  >
    <mock-toolbar
      variant="dense"
    >
      <div
        class="makeStyles-navbarContainer-10"
      >
        <div
          class="makeStyles-leftSideContainer-11"
        >
          <div
            class="makeStyles-titleContainer-12"
          >
            <img
              alt="logo"
              class="makeStyles-logo-1"
              src="images/moz-flag-inam-logo.png"
            />
            <div
              class="MuiBox-root MuiBox-root-15"
              style="display: flex; flex-direction: column;"
            >
              <mock-typography
                classname="makeStyles-title-4"
                color="secondary"
                variant="h6"
              >
                PRISM
              </mock-typography>
            </div>
          </div>
          <div
            class="makeStyles-panelsContainer-13"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              style="background-color: rgb(99, 178, 189); color: rgb(16, 16, 16);"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 18.54l-7.37-5.73L3 14.07l9 7 9-7-1.63-1.27zM12 16l7.36-5.73L21 9l-9-7-9 7 1.63 1.27L12 16zm0-11.47L17.74 9 12 13.47 6.26 9 12 4.53z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              style="color: white;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5 9.2h3V19H5V9.2zM10.6 5h2.8v14h-2.8V5zm5.6 8H19v6h-2.8v-6z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              style="color: white;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16 10h-2v2h2v-2zm0 4h-2v2h2v-2zm-8-4H6v2h2v-2zm4 0h-2v2h2v-2zm8-6H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V6h16v12z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              style="color: white;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.07 1.01h-6v2h6v-2zm-4 13h2v-6h-2v6zm8.03-6.62l1.42-1.42c-.43-.51-.9-.99-1.41-1.41l-1.42 1.42C16.14 4.74 14.19 4 12.07 4c-4.97 0-9 4.03-9 9s4.02 9 9 9 9-4.03 9-9c0-2.11-.74-4.06-1.97-5.61zm-7.03 12.62c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <mock-menu
              open="false"
            >
              <mock-menuitem
                selected="false"
              >
                A. Action Drought
              </mock-menuitem>
              <mock-menuitem
                selected="false"
              >
                A. Action Storm
              </mock-menuitem>
            </mock-menu>
            <mock-button
              starticon="[object Object]"
            >
              <mock-typography
                style="color: rgb(255, 255, 255); text-transform: none;"
              >
                Go To
              </mock-typography>
            </mock-button>
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              style="color: white;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9z"
                  />
                  <circle
                    cx="12"
                    cy="9"
                    r="2.5"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <mock-menu
              keepmounted="true"
              open="false"
            />
          </div>
        </div>
        <div
          class="makeStyles-rightSideContainer-14"
        >
          <mock-button
            classname="makeStyles-triggerButton-23"
            starticon="[object Object]"
            style="background-color: rgb(99, 178, 189);"
          >
            <mock-typography
              style="color: rgb(16, 16, 16); text-transform: none;"
            >
              Legend
            </mock-typography>
          </mock-button>
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            style="background-color: rgb(99, 178, 189);"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-icon-25"
                focusable="false"
                style="color: rgb(16, 16, 16);"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 6c3.79 0 7.17 2.13 8.82 5.5-.59 1.22-1.42 2.27-2.41 3.12l1.41 1.41c1.39-1.23 2.49-2.77 3.18-4.53C21.27 7.11 17 4 12 4c-1.27 0-2.49.2-3.64.57l1.65 1.65C10.66 6.09 11.32 6 12 6zm-1.07 1.14L13 9.21c.57.25 1.03.71 1.28 1.28l2.07 2.07c.08-.34.14-.7.14-1.07C16.5 9.01 14.48 7 12 7c-.37 0-.72.05-1.07.14zM2.01 3.87l2.68 2.68C3.06 7.83 1.77 9.53 1 11.5 2.73 15.89 7 19 12 19c1.52 0 2.98-.29 4.32-.82l3.42 3.42 1.41-1.41L3.42 2.45 2.01 3.87zm7.5 7.5l2.61 2.61c-.04.01-.08.02-.12.02-1.38 0-2.5-1.12-2.5-2.5 0-.05.01-.08.01-.13zm-3.4-3.4l1.75 1.75c-.23.55-.36 1.15-.36 1.78 0 2.48 2.02 4.5 4.5 4.5.63 0 1.23-.13 1.77-.36l.98.98c-.88.24-1.8.38-2.75.38-3.79 0-7.17-2.13-8.82-5.5.7-1.43 1.72-2.61 2.93-3.53z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <mock-list
            classname="makeStyles-list-24"
            disablepadding="true"
          />
          <mock-printimage />
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root"
            href="https://github.com/wfp-VAM/prism-app"
            style="color: white;"
            tabindex="0"
            target="_blank"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-github "
                data-icon="github"
                data-prefix="fab"
                focusable="false"
                font-size="1.5rem"
                role="img"
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <mock-button
            component="a"
            href="https://innovation.wfp.org/project/prism"
            starticon="[object Object]"
            style="font-size: 1.25rem;"
            target="_blank"
          >
            <mock-typography
              color="secondary"
              style="text-transform: none;"
            >
              About
            </mock-typography>
          </mock-button>
          <mock-button
            aria-label="language-select-dropdown-button"
            endicon="[object Object]"
            style="padding-left: 0px;"
          >
            <mock-typography
              color="secondary"
              style="text-transform: none;"
            >
              en
            </mock-typography>
          </mock-button>
          <mock-menu
            classname="makeStyles-block-26"
            open="false"
          >
            <mock-menuitem>
              <mock-typography>
                en
              </mock-typography>
            </mock-menuitem>
            <mock-menuitem>
              <mock-typography>
                pt
              </mock-typography>
            </mock-menuitem>
          </mock-menu>
        </div>
      </div>
    </mock-toolbar>
  </mock-appbar>
</div>
`;
